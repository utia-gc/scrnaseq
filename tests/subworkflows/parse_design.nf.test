nextflow_workflow {

    name "Test Workflow Parse_Design"
    script "subworkflows/parse_design.nf"
    workflow "Parse_Design"

    test("Parse_Design successfully creates read channels from paired-end samplesheet.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/utia-gc/ngs-test/raw/scrnaseq/data/samplesheets/samplesheet_L001_L002.csv"
                """
            }
        }

        then {
            // test workflow status
            assert workflow.success

            // test creates expected channels of samples
            with(workflow.out.samples) {
                assert size() == 2

                // test lane 1
                LinkedHashMap expectedMetadataLane1 = [
                    sampleName:   'Chromium_3p_GEX_Human_PBMC_chr21_100000',
                    sampleNumber: '1',
                    lane:         '001',
                    readType:     'paired'
                ]
                with(get(0)) {
                    assert get(0) == expectedMetadataLane1
                }

                // test lane 2
                LinkedHashMap expectedMetadataLane2 = [
                    sampleName:   'Chromium_3p_GEX_Human_PBMC_chr21_100000',
                    sampleNumber: '1',
                    lane:         '002',
                    readType:     'paired'
                ]
                with(get(1)) {
                    assert get(0) == expectedMetadataLane2
                }
            }

            // test output channels reproducibility
            assert snapshot(workflow.out).match()
        }

    }

}
