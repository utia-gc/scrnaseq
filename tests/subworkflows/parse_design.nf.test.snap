{
    "successfully creates channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "successfully creates read channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "Parse_Design successfully creates read channels from single-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "Parse_Design successfully creates read channels from mixed single-end and paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "successfully creates read channels from paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "successfully creates read channels from mixed single-end and paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924569_GSM3073206_Saccharomyces_cerevisiae-AR_Biological_Repeat-2_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1",
                            "readType": "paired"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz",
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "SRR1066658_GSM1299414_WT_NR_B",
                            "readType": "single"
                        },
                        [
                            "/trev-f/SRAlign-test/raw/rnaseq/data/reads/SRR1066658_GSM1299414_WT_NR_B_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    },
    "Parse_Design successfully creates read channels from paired-end samplesheet.": {
        "content": [
            {
                "0": [
                    [
                        {
                            "sampleName": "Chromium_3p_GEX_Human_PBMC_chr21_100000",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired"
                        },
                        [
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L001_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L001_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "Chromium_3p_GEX_Human_PBMC_chr21_100000",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired"
                        },
                        [
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L002_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L002_R2_001.fastq.gz"
                        ]
                    ]
                ],
                "samples": [
                    [
                        {
                            "sampleName": "Chromium_3p_GEX_Human_PBMC_chr21_100000",
                            "sampleNumber": "1",
                            "lane": "001",
                            "readType": "paired"
                        },
                        [
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L001_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L001_R2_001.fastq.gz"
                        ]
                    ],
                    [
                        {
                            "sampleName": "Chromium_3p_GEX_Human_PBMC_chr21_100000",
                            "sampleNumber": "1",
                            "lane": "002",
                            "readType": "paired"
                        },
                        [
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L002_R1_001.fastq.gz",
                            "/utia-gc/ngs-test/raw/scrnaseq/data/reads/raw/Chromium_3p_GEX_Human_PBMC_chr21_100000_S1_L002_R2_001.fastq.gz"
                        ]
                    ]
                ]
            }
        ],
        "timestamp": "2023-08-22T15:41:59+0000"
    }
}